### PR TITLE
Ruby support

### LINKED ISSUE #140: Request for Ruby support
Opening an issue to gauge interest in making ruby a supported lanugage

### PR DESCRIPTION
**Pull request by michelvocks**
Fixes #140

### CODE REVIEW

**Comment by michelvocks** [comment]
@Skarlso This ready for review.

Documentation: https://github.com/gaia-pipeline/docs/pull/9
Ruby example pipeline: https://github.com/gaia-pipeline/ruby-example

**Comment by Skarlso** [comment]
@michelvocks Ack. I'll give this a spin today. :)

**Comment by Skarlso** [comment]
@michelvocks is the go.mod file update intentional?

File: plugin/plugin.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/gaia-pipeline/gaia"
 	"github.com/gaia-pipeline/gaia/security"
 	"github.com/gaia-pipeline/gaia/workers/scheduler"
-	"github.com/gaia-pipeline/protobuf"
+	proto "github.com/gaia-pipeline/protobuf"
```
Comment (Skarlso) [inline plugin/plugin.go]: Huh? :O

**Comment by michelvocks** [comment]
> @michelvocks is the go.mod file update intentional?

Nope. No clue what is going on there. It has been magically added by go modules.

File: plugin/plugin.go (hunk 1 above)
Comment (michelvocks) [reply plugin/plugin.go]: This has been auto corrected by `goimports`. Shouldn't be a problem in my opinion.

**Comment by michelvocks** [comment]
@Skarlso I think the problem is https://github.com/golang/go/issues/27643

I didn't know that `go get` will force go mod to automatically add it as dependency. 
I've removed all indirect dependencies.

**Comment by Skarlso** [comment]
I played around with this yesterday for a while. Looks good. I had some initial trouble getting bundler to work. I wonder if we should add --vendor into the doc. But I guess however is using Ruby probably already knows these tricks.

I have one last go on the code today then it's good to merge.

**Comment by Skarlso** [comment]
@michelvocks  Do you need the name thingy for this in this? Or can we close this? It should be working. :)

**Comment by Skarlso** [comment]
@michelvocks Haha yes. I'm an idiot. Sorry. :) Go ahead. :)

**Pull request merged**